stglib/mc.py

```diff
@@ -0,0 +1,188 @@
+import pandas as pd
+import xarray as xr
+
+from .core import qaqc, utils
+
+
+def read_asc(filnam, skiprows=50, encoding="utf-8"):
+    """Read data from an SBE 37 MicroCAT .asc file into an xarray
+    Dataset.
+
+    Parameters
+    ----------
+    filnam : string
+        The filename
+    skiprows : int, optional
+        How many header rows to skip. Default 50
+    encoding : string, optional
+        File encoding. Default 'utf-8'
+    Returns
+    -------
+    xarray.Dataset
+        An xarray Dataset of the MicroCAT data
+    """
+
+    df = pd.read_csv(
+        filnam,
+        skiprows=skiprows,
+        header=None,
+        names=["Temp", "Cond", "Sal", "Date", "Time"],
+        parse_dates={"time": ["Date", "Time"]},
+        encoding=encoding,
+        index_col=False,
+    )
+    print(df)
+    df.index.names = ["time"]
+    mc = xr.Dataset.from_dataframe(df)
+    return mc
+
+
+# Make raw CDF
+def asc_to_cdf(metadata):
+    basefile = metadata["basefile"]
+
+    ds = read_asc(basefile + ".asc", skiprows=metadata["skiprows"])
+
+    metadata.pop("skiprows")
+
+    ds = utils.write_metadata(ds, metadata)
+
+    ds = utils.ensure_cf(ds)
+
+    # configure file
+    cdf_filename = ds.attrs["filename"] + "-raw.cdf"
+
+    ds.to_netcdf(cdf_filename, unlimited_dims=["time"])
+
+    print("Finished writing data to %s" % cdf_filename)
+
+    return ds
+
+
+# Process data and write to .nc file
```

Comment:
Remove this comment and/or add to the existing comment immediately below the function definition.